### ISSUE TITLE
gromacs does not work if nix-shell is activated with local nix file 

### ISSUE THREAD

**Issue by fperakis** [fperakis/MD-nix_glycerol_water#1]
@markuskowa 
This works: first `nix-shell -p qchem-unstable.gromacs` and then ` gmx -version`

If I use `nix-shell nix_files/cpu.nix` it opens a new nix-shell but it does not have gromacs, since if I use  ` gmx -version` I get `bash: gmx: command not found`. The same goes for `nix-shell nix_files/cuda.nix`

**Comment by markuskowa** [comment]
The CPU and CUDA nix files use the MPI version. This should work: `gmx_mpi -version`. We also made it a little bit more generic by export `GMX`, so this should also work: `$GMX -version`.
[See](https://github.com/fperakis/MD-nix_glycerol_water/blob/c19f0a2c006fe80436c25728bcf528e769b266bc/nix_files/cuda.nix#L40)

**Comment by fperakis** [comment]
Good point, I tried `gmx_mpi -version` and gives the following output
`[sol-nix:30132:0:30132] Caught signal 4 (Illegal instruction: illegal operand)
==== backtrace (tid:  30132) ====
 0 0x0000000000012ef0 __funlockfile()  ???:0
 1 0x000000000067df83 gmx::erfinv()  ???:0
 2 0x0000000000204725 _GLOBAL__sub_I_tabulatednormaldistribution.cpp()  ???:0
 3 0x000000000000fbce call_init()  ???:0
 4 0x000000000000fcb4 _dl_init()  ???:0
 5 0x000000000000108a _dl_start_user()  ???:0
====================`

**Comment by fperakis** [comment]
the command `which gmx_mpi` gives also
`/nix/store/4593jrvygsb5s57kh5ijkh55ca8c6wsk-gromacs-2021.4/bin/gmx_mpi`

**Comment by fperakis** [comment]
Note also that there is a difference in versions:

- in a `nix-shell -p qchem-unstable.gromacs ` terminal the command `which gmx` gives `/nix/store/14b5q26714xqxwdzn40m1z9mbmr8d0pa-gromacs-2022.4/bin/gmx`

- whereas in a `nix-shell nix_files/cpu.nix` terminal the  command `which gmx_mpi` gives `/nix/store/4593jrvygsb5s57kh5ijkh55ca8c6wsk-gromacs-2021.4/bin/gmx_mpi`

**Comment by markuskowa** [comment]
> Good point, I tried `gmx_mpi -version` and gives the following output `[sol-nix:30132:0:30132] Caught signal 4 (Illegal instruction: illegal operand) ==== backtrace (tid: 30132) ==== 0 0x0000000000012ef0 __funlockfile() ???:0 1 0x000000000067df83 gmx::erfinv() ???:0 2 0x0000000000204725 _GLOBAL__sub_I_tabulatednormaldistribution.cpp() ???:0 3 0x000000000000fbce call_init() ???:0 4 0x000000000000fcb4 _dl_init() ???:0 5 0x000000000000108a _dl_start_user() ???:0 ====================`

I see, this will only run on a compute node. The login node has an older CPU that support the optmizations.

**Comment by markuskowa** [comment]
> Note also that there is a difference in versions:
> 
> * in a `nix-shell -p qchem-unstable.gromacs ` terminal the command `which gmx` gives `/nix/store/14b5q26714xqxwdzn40m1z9mbmr8d0pa-gromacs-2022.4/bin/gmx`
> * whereas in a `nix-shell nix_files/cpu.nix` terminal the  command `which gmx_mpi` gives `/nix/store/4593jrvygsb5s57kh5ijkh55ca8c6wsk-gromacs-2021.4/bin/gmx_mpi`

Yep, this is intended:  `qchem-unstable.gromacs` give you the latest and greatest. In `nix_files/*.nix` we have pinned the versions to keep it reproducible (we should update the versions in the nix files to get best out of it).

**Comment by fperakis** [comment]
> I see, this will only run on a compute node. The login node has an older CPU that support the optmizations.

I should have a workaround by doing the preprocessing on a compute node.
